core/src/main/scala/IndexedItemized.scala

```diff
@@ -21,9 +21,11 @@ trait IndexedItemized extends Itemized {
 }
 
 trait ItemizedIndex[T <: IndexedItemized] {
-  def toIndex(c: T): T#Index
+  def toIndex(c: T): T#Index = c.index
 
-  def fromIndex(v: T#Index): Option[T]
+  def fromIndex(v: T#Index): Option[T] = indexMap.get(v)
+
+  val indexMap: Map[T#Index, T]
```

Comment:
@esarbe, what do you think of something like this? You can get `.keys`, `.values` and `.size` from the map.